src/main/scala/is/hail/utils/richUtils/RichIterableGenotype.scala

```diff
@@ -0,0 +1,8 @@
+package is.hail.utils.richUtils
+
+import is.hail.utils.{IntIterator, HardCallIterator}
+import is.hail.variant.Genotype
+
+class RichIterableGenotype(it: Iterable[Genotype]) {
+  def hardCallIterator: HardCallIterator = new HardCallIterator(it.iterator)
```

Comment:
You can't dynamically choose to resolve through an implicit vs a virtual function.  In particular, Iterable[Int] doesn't have a hardCallIterator method so it will call it, even if it is pointing to a subclass that does.

Reusing the same name does could cause confusion.  You might rename the method on GenotypeStream.

I think we normally reserve `it` for `Iterator`s.

You can instantiate a subclass without naming it.

```
class RichIterableGenotype(gs: Iterable[Genotype]) {
  def hardCallIterator: IntIterator = gs match {
    case gs: GenotypeStream => gs.hardCallIterator
    case _ =>
      new IntIterator {
        val it = gs.iterator
        override def hasNext: Boolean = it.hasNext
        override def nextInt(): Int = it.next().unboxedGT
      }
  }
}
```